File: expensereport-fortran/ExpenseReport.f
      MODULE ExpenseReport
        IMPLICIT NONE
        ENUM, BIND(C)
          ENUMERATOR :: BREAKFAST, DINNER, CAR_RENTAL
        ENDENUM

        TYPE :: expense
          INTEGER :: type
          INTEGER :: amount
        END TYPE

        CONTAINS
        SUBROUTINE printReport(htmlMode, expenses)
          INTEGER :: htmlMode
          TYPE(expense), DIMENSION (:), ALLOCATABLE :: expenses
          INTEGER :: total = 0
          INTEGER :: mealExpenses = 0
          INTEGER :: i
          CHARACTER(LEN = 20) :: expenseName
          CHARACTER(LEN = 1) :: mealOverExpensesMarker
          IF (htmlMode == 1) THEN
            PRINT "(a)", '<!DOCTYPE html>'
            PRINT "(a)", '<html lang="en">'
            PRINT "(a)", '<head>'
            PRINT "(a)", '<title>Expense Report</title>'
            PRINT "(a)", '</head>'
            PRINT "(a)", '<body>'
            PRINT "(a)", '<h1>Expense Report</h1>'
          ELSE
            PRINT "(a)", 'Expense Report'
          END IF

          IF (htmlMode == 1) THEN
            PRINT "(a)", '<table>'
            PRINT "(a)", '<thead>'
            PRINT "(a)", '<tr>'
            PRINT "(a)", '<th scope="col">Type</th>'
            PRINT "(a)", '<th scope="col">Amount</th>'
            PRINT "(a)", '<th scope="col">Over Limit</th>'
            PRINT "(a)", '</tr>'
            PRINT "(a)", '</thead>'
            PRINT "(a)", '<tbody>'
          END IF
          DO i = LBOUND(expenses, 1), UBOUND(expenses, 1)
            IF (expenses(i)%type == BREAKFAST .OR. expenses(i)%type == DINNER) &
              mealExpenses = mealExpenses + expenses(i)%amount
            SELECT CASE (expenses(i)%type)
              CASE (DINNER)
                expenseName = "Dinner"
              CASE (BREAKFAST)
                expenseName = "Breakfast"
              CASE (CAR_RENTAL)
                expenseName = "Car Rental"
            END SELECT
            IF (expenses(i)%type == BREAKFAST .AND. expenses(i)%amount > 1000 .OR. &
                expenses(i)%type == DINNER .AND. expenses(i)%amount > 5000) THEN
              mealOverExpensesMarker = "X"
            ELSE
              mealOverExpensesMarker = " "
            END IF
            total = total + expenses(i)%amount
            IF (htmlMode == 1) THEN
              PRINT "(a,a,a,1x,i10,a,1x,a,a)", &
                '<tr><td>', &
                expenseName, &
                '</td><td>', &
                expenses(i)%amount, &
                '</td><td>', &
                mealOverExpensesMarker, &
                '</td></tr>'
            ELSE
              PRINT "(a,1x,i10,1x,a)", expenseName, expenses(i)%amount, mealOverExpensesMarker
            END IF
          END DO
          IF (htmlMode == 1) THEN
            PRINT "(a)", '</tbody>'
            PRINT "(a)", '</table>'
          END IF

          IF (htmlMode == 1) THEN
            PRINT "(a,i10,a)", '<p>Meal expenses: ', mealExpenses, &
              '</p>'
            PRINT "(a,i10,a)", '<p>Total: ', total, '</p>'
          ELSE
            PRINT "(a,i10)", 'Meal expenses: ', mealExpenses
            PRINT "(a,i10)", 'Total: ', total
          END IF

          IF (htmlMode == 1) THEN
            PRINT "(a)", '</body>'
            PRINT "(a)", '</html>'
          END IF
        END SUBROUTINE printReport

      END MODULE ExpenseReport

      PROGRAM main
        USE ExpenseReport
        IMPLICIT NONE
        TYPE(expense), DIMENSION (:), ALLOCATABLE :: expenses
        ALLOCATE(expenses(5))
        expenses(1)%type = BREAKFAST
        expenses(1)%amount = 1000
        expenses(2)%type = BREAKFAST
        expenses(2)%amount = 1001
        expenses(3)%type = DINNER
        expenses(3)%amount = 5000
        expenses(4)%type = DINNER
        expenses(4)%amount = 5001
        expenses(5)%type = CAR_RENTAL
        expenses(5)%amount = 4
        CALL printReport(1, expenses)
        DEALLOCATE(expenses)
      END PROGRAM main

File: expensereport-php/ExpenseReport.php
<?php

abstract class ExpenseType {
    const DINNER = 1;
    const BREAKFAST = 2;
    const CAR_RENTAL = 3;
}

class Expense {
    public $type;
    public $amount;
    function __construct($type, $amount) {
        $this->type = $type;
        $this->amount = $amount;
    }
}

class ExpenseReport {
    public function print_report($htmlMode, $expenses) {
        $mealExpenses = 0;
        $total = 0;
        $date = date("Y-m-d h:i:sa");
        if ($htmlMode) {
            ?>
<!DOCTYPE html>
<html lang="en">
<head>
<title>Expense Report <?= $date?></title>
</head>
<body>
<h1>Expense Report <?= $date?></h1>
<?php
        } else {
            print("Expense Report {$date}\n");
        }
        if ($htmlMode) {
            ?>
<table>
<thead>
<tr><th scope="col">Type</th><th scope="col">Amount</th><th scope="col">Over Limit</th></tr>
</thead>
<tbody>
<?php
        }
        foreach ($expenses as $expense) {
            if ($expense->type == ExpenseType::DINNER || $expense->type == ExpenseType::BREAKFAST) {
                $mealExpenses += $expense->amount;
            }
            $expenseName = "";
            switch ($expense->type) {
                case ExpenseType::DINNER: $expenseName = "Dinner"; break;
                case ExpenseType::BREAKFAST: $expenseName = "Breakfast"; break;
                case ExpenseType::CAR_RENTAL: $expenseName = "Car Rental"; break;
            }

            $mealOverExpensesMarker = $expense->type == ExpenseType::DINNER && $expense->amount > 5000 || $expense->type == ExpenseType::BREAKFAST && $expense->amount > 1000 ? "X" : " ";
            if ($htmlMode) {
                ?>
<tr><td><?= $expenseName ?></td><td><?= $expense->amount ?></td><td><?= $mealOverExpensesMarker ?></td></tr>
<?php
            } else {
                print($expenseName . "\t" . $expense->amount . "\t" . $mealOverExpensesMarker . "\n");
            }
            $total += $expense->amount;
        }
        if ($htmlMode) { ?>
</tbody>
</table>
<?php }

        if ($htmlMode) {
            ?>
<p>Meal Expenses: <?= $mealExpenses ?></p>
<p>Total Expenses: <?= $total ?></p>
<?php } else {
            print("Meal Expenses: " . $mealExpenses . "\n");
            print("Total Expenses: " . $total . "\n");
        }

        if ($htmlMode) { ?>
</body>
</html>
<?php }
    }
}

ExpenseReport::print_report(true, [
    new Expense(ExpenseType::BREAKFAST, 1000),
    new Expense(ExpenseType::BREAKFAST, 1001),
    new Expense(ExpenseType::DINNER, 5000),
    new Expense(ExpenseType::DINNER, 5001),
    new Expense(ExpenseType::CAR_RENTAL, 4),
]);
